pymchelper/utils/mcscripter.py

```diff
@@ -0,0 +1,303 @@
+        r = []
+
+        if "${" in s:
+            _i = [i for i, d in enumerate(s) if d == "{"]
+            _ii = [i for i, d in enumerate(s) if d == "}"]
+
+            for i in zip(_i, _ii):
+                r.append(s[i[0] + 1:i[1]])
+        return r
+
+    @staticmethod
+    def lreplace(s, f, r):
+        """
+        Left adjusted replacement of string f with string r, in string s.
+
+        This function is implemented in order to fill in data in FORTRAN77 fields,
+        which are tied to certain positions on the line, i.e. subsequent values
+        may not be shifted.
+
+        Finds f in s and replaces it with r, but left adjusted, retaining line length.
+        If r is shorter than f, remaining chars will be space padded.
+        If r is larger than f, then characters will be overwritten.
+        A copy of s with the replacement is returned.
+        """
+        if f in s:
+            _idx = s.find(f)
+            if len(r) < len(f):
+                _r = r + " " * (len(f) - len(r))
+            else:
+                _r = r
+            text = s[:_idx] + _r + s[_idx + len(_r):]
+            return text
+
+    def write(self, t, u_dict):
+        """
+        Write a copy of the template, using the substitutions as specifed in
+        the unique dictionary u_dict.
+
+        "Unique", means that any _MIN _MAX _STEP type variables have been set.
+        """
+
+        _wd = u_dict["WDIR"]
+
+        # check if any keys are in WDIR subsitutions
+        for key in u_dict.keys():
+            token = "${" + key + "}"
+            if token in _wd:
+
+                _s = u_dict[key]
+                if isinstance(_s, float):
+                    _s = "{:08.3f}".format(u_dict[key])  # HARDCODED float format for directory string
+                _wd = _wd.replace(token, _s)
+        wdir = _wd
+
+        for tf in t.files:  # tf = template filename
+            of = McFile()  # of = output file object
+            of.fname = tf.fname
+            of.path = os.path.join(wdir, tf.fname)
+            of.tdir = tf.tdir
+
+            # symlinks should not be parsed for tokens.
+            if tf.symlink:
+                of.symlink = True
+            else:
+                of.symlink = False
+
+                for line in tf.lines:
+                    for key in u_dict.keys():
+                        token = "${" + key + "}"
+                        while token in line:
+                            _s = u_dict[key]
+                            if isinstance(_s, float):
+                                _s = "{:.3f}".format(u_dict[key])  # HARDCODED float format for loop_keys
+                            line = self.lreplace(line, token, _s)
+                    of.lines.append(line)
+            # end loop over t.files
+            of.write()
+
+
+def main(args):
+    """
+    Main function.
+    """
+    logger.setLevel('INFO')
+
+    if len(args) < 1:
+        print("usage: {} <configfile>".format("mcscripter.py"))
```

Comment:
This won't be valid if you use an executable produced by setup.py. 
(`mcscripter` no `mcscripter.py`)
Consider some simple ArgParse setup, like here
https://github.com/DataMedSci/pymchelper/blob/master/pymchelper/utils/pld2sobp.py#L123

Reply:
will add it, wanted that anyway